tools/system_libs.py

```diff
@@ -812,6 +802,27 @@ def get_files(self):
     return libc_files
 
 
+class libc_environ(libc):
+  name = 'libc_environ'
+  cflags = ['-Os', '-fno-builtin']
+
+  def get_files(self):
+    if shared.Settings.WASM_BACKEND:
+      # See libc_extras below
+      # Include all the getenv stuff with the wasm backend. With fastcomp we
+      # still use JS because libc is a .bc file and we don't want to have a
+      # global constructor there for __environ, which would mean it is always
+      # included.
+      return files_in_path(
```

Comment:
Can probably drop this conditional and move the comment to the `can_build` method.